SingularityService/src/main/java/com/hubspot/singularity/data/LoggingCuratorFramework.java

```diff
@@ -0,0 +1,323 @@
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import org.apache.curator.CuratorZookeeperClient;
+import org.apache.curator.framework.CuratorFramework;
+import org.apache.curator.framework.WatcherRemoveCuratorFramework;
+import org.apache.curator.framework.api.CreateBuilder;
+import org.apache.curator.framework.api.CuratorListener;
+import org.apache.curator.framework.api.DeleteBuilder;
+import org.apache.curator.framework.api.ExistsBuilder;
+import org.apache.curator.framework.api.GetACLBuilder;
+import org.apache.curator.framework.api.GetChildrenBuilder;
+import org.apache.curator.framework.api.GetConfigBuilder;
+import org.apache.curator.framework.api.GetDataBuilder;
+import org.apache.curator.framework.api.ReconfigBuilder;
+import org.apache.curator.framework.api.RemoveWatchesBuilder;
+import org.apache.curator.framework.api.SetACLBuilder;
+import org.apache.curator.framework.api.SetDataBuilder;
+import org.apache.curator.framework.api.SyncBuilder;
+import org.apache.curator.framework.api.UnhandledErrorListener;
+import org.apache.curator.framework.api.transaction.CuratorMultiTransaction;
+import org.apache.curator.framework.api.transaction.CuratorTransaction;
+import org.apache.curator.framework.api.transaction.TransactionOp;
+import org.apache.curator.framework.imps.CuratorFrameworkState;
+import org.apache.curator.framework.listen.Listenable;
+import org.apache.curator.framework.schema.SchemaSet;
+import org.apache.curator.framework.state.ConnectionStateErrorPolicy;
+import org.apache.curator.framework.state.ConnectionStateListener;
+import org.apache.curator.utils.EnsurePath;
+import org.apache.zookeeper.Watcher;
+import org.apache.zookeeper.server.quorum.flexible.QuorumVerifier;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class LoggingCuratorFramework implements CuratorFramework {
+  private final CuratorFramework curator;
+
+  private final ConcurrentMap<String, Integer> counters;
+
+  private final Long interval = 3L;
+
+  private static final Logger LOG = LoggerFactory.getLogger(
+    LoggingCuratorFramework.class
+  );
+
+  private final ScheduledExecutorService executorService;
+
+  private class ClearCounter implements Runnable {
+
+    @Override
+    public void run() {
+      try {
+        logAndClear();
+      } catch (Exception e) {
+        LOG.error("Failed to log and clear ZooKeeper calls", e);
+      }
+    }
+  }
+
+  @Inject
+  public LoggingCuratorFramework(
+    CuratorFramework curator,
+    SingularityManagedScheduledExecutorServiceFactory executorServiceFactory
+  ) {
+    this.curator = curator;
+    counters = new ConcurrentHashMap<>();
+
+    this.executorService = executorServiceFactory.get("logging-curator-framework");
+    this.executorService.scheduleAtFixedRate(
+        new ClearCounter(),
```

Comment:
slightly different way of doing this in functional form. This can be just a private method instead of a class that extends runnable and you can pass the method reference to the executor service call.
edit: and add the try/catch in the private method then as well